### ISSUE TITLE
Error when adding new events from web interface (works from App)

### ISSUE THREAD

**Issue by Gordontgopher** [home-assistant/frontend#14674]
### The problem

When I try to add an event to my local calendar I get an error;
`2022-12-09 11:19:51.976 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [547538714528] Error handling message: Could not parse date for dictionary value @ data['event']['dtend']. Got 'Invalid DateTInvalid Date'
Could not parse date for dictionary value @ data['event']['dtstart']. Got 'Invalid DateTInvalid Date' (invalid_format) from 192.168.1.106 (Mozilla/5.0 (X11; CrOS x86_64 14541.0.0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/110.0.0.0 Safari/537.36)`

Adding events from the Android app (logged on as same user)  is successful. App is beta-2982-956b2cc-full

### What version of Home Assistant Core has the issue?

2022.12.1

### What type of installation are you running?

Home Assistant Supervised

### Integration causing the issue

Local Calendar

### Link to integration documentation on our website

https://www.home-assistant.io/integrations/local_calendar/

### Example YAML snippet

```yaml
N/A
```

### Anything in the logs that might be useful for us?

```txt
2022-12-09 11:19:51.976 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [547538714528] Error handling message: Could not parse date for dictionary value @ data['event']['dtend']. Got 'Invalid DateTInvalid Date'
Could not parse date for dictionary value @ data['event']['dtstart']. Got 'Invalid DateTInvalid Date' (invalid_format) from 192.168.1.106 (Mozilla/5.0 (X11; CrOS x86_64 14541.0.0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/110.0.0.0 Safari/537.36)
```

### Additional information

System configured for GMT, Language is English (GB) and time format is Auto (use language setting). Error also occurs with all other options for user time format.

**Comment by spacegaier** [comment]
@Gordontgopher You stated it works with the Android app, so that means where does it not work for you (browser and OS)? What are the inputs that you've put into the creation dialog form?

**Comment by SkeLLLa** [comment]
Same issue. 

Browser: Firefox 107
OS: linux

Top stacktrace
```
Uncaught (in promise) RangeError: date value is not finite in DateTimeFormat.format()
    c a8b2ec7d.js:1
    value ha-date-input.ts:59
    update app.f8884e15.js:1366
    performUpdate app.f8884e15.js:1366
    scheduleUpdate app.f8884e15.js:1366
    _$EC app.f8884e15.js:1366
    requestUpdate app.f8884e15.js:1366
    o app.f8884e15.js:1366
    b app.f8884e15.js:1366
    w app.f8884e15.js:1366
    F ha-chip.ts:85
```
![image](https://user-images.githubusercontent.com/2273103/206840968-81c3588c-016e-403d-9ac0-18ca6d23f67c.png)

**Comment by spacegaier** [comment]
@SkeLLLa Sound to me like yours is a separate topic, because OP did not mention anything about missing input fields. Also you posted a frontend browser stacktrace, while OP posted one from the backend.

Are you on the same version as OP because we had fixed some issues with the date selector in 2022.12.1.

**Comment by SkeLLLa** [comment]
@spacegaier I'm also on 2022.12.1

**Comment by Gordontgopher** [comment]
> @Gordontgopher You stated it works with the Android app, so that means where does it not work for you (browser and OS)? What are the inputs that you've put into the creation dialog form?

![Screenshot 2022-12-11 13 57 47](https://user-images.githubusercontent.com/3321366/206908111-9a15fb23-bbd9-4257-b3a4-44658034a7f8.png)

**Comment by SkeLLLa** [comment]
Updated the HA to 2022.12.4 with frontend version 20221212.0 and the bug on Firefox desktop is gone for me.

**Comment by allenporter** [comment]
I'm assuming this is related to my change to `ha-date` to append a timestamp to the date.

**Comment by allenporter** [comment]
Are you folks using any particular browser language or locale settings? I have not yet been able to reproduce this.

**Comment by SkeLLLa** [comment]
@allenporter In my case the issue was reproducible on Firefox browser on windows and linux. In both systems I have `en-US` locale set by default. However on linux for time formatting I have `LC_TIME=C` as default. On windows I guess there's a customized format for dates and time, which is based on en-US.

I've didn't change the locale, however if it helps.
In the same time when issue was present I've checked it in chrome browser on same machines and I was unable to reproduce it. In particular I've checked that on 2 laptops, running different versions of Firefox and different OS: Windows11, Ubuntu and Gentoo linux. In all cases chrome worked, firefox not. But after update of frontend to 20221212.0 this issue gone for me on all devices.
